index.ts

```diff
@@ -58,6 +58,12 @@ export const setupPlugin: BigQueryPlugin['setupPlugin'] = async (meta) => {
 
     global.bigQueryTable = global.bigQueryClient.dataset(config.datasetId).table(config.tableId)
 
+    const existingFields = await meta.cache.get('bigQueryTableFieldsSynced', 0)
```

Comment:
non-blocking

Maybe worth just caching the whole metadata? Or a hash of it or something? Although I guess we're always just adding fields not changing anything so maybe doesn't add much.

Reply:
I don't think it adds anything to do so besides potential serialization woes. 

If we need to change the structure in a way which this code doesn't support we need to do cache-busting anyways.

Reply:
I would argue that this is important. Specifically if someone changes their BQ configuration to write to a different table or dataset and we still haven't expired the cache yet, then we wouldn't set that up.

Less importantly if we replace a field we might miss the update, but we can address that once we start changing fields, e.g. by caching fields version used.

Reply:
Agreed also said this for the second part (with "we" I meant anyone changing the code for this plugin):
>  but we can address that once we start changing fields